moveit_ros/moveit_servo/src/servo_parameters.cpp

```diff
@@ -77,74 +78,259 @@ ServoParameters::setParametersCallback(std::vector<rclcpp::Parameter> parameters
+                                     parameter_set::ParameterDescriptorBuilder()
+                                         .type(rclcpp::ParameterType::PARAMETER_DOUBLE)
+                                         .description("Stop when the condition number hits this"));
+  node_parameters->declare_parameter(
+      ns + ".joint_limit_margin", rclcpp::ParameterValue(parameters->joint_limit_margin),
+      parameter_set::ParameterDescriptorBuilder()
+          .type(rclcpp::ParameterType::PARAMETER_DOUBLE)
+          .description("Added as a buffer to joint limits [radians]. If moving quickly, make this larger."));
 
-  // Get the parameters (organized same order as YAML file)
-  declareOrGetParam<bool>(parameters->use_gazebo, ns + ".use_gazebo", node, logger);
-  declareOrGetParam<std::string>(parameters->status_topic, ns + ".status_topic", node, logger);
+  // Collision checking
+  node_parameters->declare_parameter(ns + ".check_collisions", rclcpp::ParameterValue(parameters->check_collisions),
+                                     parameter_set::ParameterDescriptorBuilder()
+                                         .type(rclcpp::ParameterType::PARAMETER_BOOL)
+                                         .description("Check collisions?"));
+  node_parameters->declare_parameter(
+      ns + ".collision_check_rate", rclcpp::ParameterValue(parameters->collision_check_rate),
+      parameter_set::ParameterDescriptorBuilder()
+          .type(rclcpp::ParameterType::PARAMETER_DOUBLE)
+          .description("[Hz] Collision-checking can easily bog down a CPU if done too often. Collision checking begins "
+                       "slowing down when nearer than a specified distance."));
+  node_parameters->declare_parameter(ns + ".self_collision_proximity_threshold",
+                                     rclcpp::ParameterValue(parameters->self_collision_proximity_threshold),
+                                     parameter_set::ParameterDescriptorBuilder()
+                                         .type(rclcpp::ParameterType::PARAMETER_DOUBLE)
+                                         .description("Start decelerating when a self-collision is this far [m]"));
+  node_parameters->declare_parameter(ns + ".scene_collision_proximity_threshold",
+                                     rclcpp::ParameterValue(parameters->scene_collision_proximity_threshold),
+                                     parameter_set::ParameterDescriptorBuilder()
+                                         .type(rclcpp::ParameterType::PARAMETER_DOUBLE)
+                                         .description("Start decelerating when a scene collision is this far [m]"));
+}
 
+void ServoParameters::get(const std::string& ns,
+                          const rclcpp::node_interfaces::NodeParametersInterface::SharedPtr& node_parameters,
+                          std::shared_ptr<ServoParameters> parameters)
```

Comment:
Instead of using an output parameter here, this should just return a new ServoParameters object.  Something like this should work:

```suggestion
ServoParameters ServoParameters::get(const std::string& ns,
                          const rclcpp::node_interfaces::NodeParametersInterface::SharedPtr& node_parameters)
```

Then since this function is a member of the class it should be able to use the private constructor.